paramiko/sftp_client.py

```diff
@@ -704,20 +846,26 @@ def putfo(self, fl, remotepath, file_size=0, callback=None, confirm=True):
         :param bool confirm:
             whether to do a stat() on the file afterwards to confirm the file
             size (since 1.7.7)
+        :param str encoding:
+            encoding for file paths/names and content (default: utf8)
```

Comment:
The `encoding` and `errors` parameters are not used for the content because `putfo` uses binary mode.

Reply:
I guess you refer to the following part:
```
with self.file(
            remotepath, "wb", encoding=encoding, errors=errors
        ) as fr:
            fr
```
Right, it's in binary mode, but that's just the content, the file path might be non uft8 though. And just below that part the stat method is also used, which takes these parameters to get the actual remote path in the remote encoding.

Actually, this remotepath over there might need encoding too. I'm not sure... I'll try and test this week. (All you need is a VM or Docker with an OS using an encoding different from utf8...)
-> `remotepath.decode("utf-8", errors).encode(encoding, errors)`

Reply:
Update:    
self.file is an object of open and open takes encoding and errors as parameter. Encoding and errors in open decodes the file name/path.

In other words: Everything should be correct here. You didn't dig deep enough, I guess.

Reply:
What I mean is that the comment for the parameter should be:
```
encoding for file paths/names (default: utf8)
```
I think the current comment leads misleading. In fact, the encoding parameter is not used for the content.

Reply:
I fixed that now.